code/game/machinery/telecomms/telecomunications.dm

```diff
@@ -100,6 +110,15 @@ GLOBAL_LIST_EMPTY(telecomms_list)
 	links = list()
 	return ..()
 
+/obj/machinery/telecomms/proc/get_temperature()
+	return GetComponent(/datum/component/server).temperature
+
+/obj/machinery/telecomms/proc/get_efficiency()
+	return GetComponent(/datum/component/server).efficiency
+
+/obj/machinery/telecomms/proc/get_overheat_temperature()
+	return GetComponent(/datum/component/server).overheated_temp
```

Comment:
Byond can't handle this kind of practice, afaik. You need a type holder to access the value.

Reply:
done